src/renderers/shared/fiber/ReactFiberReconciler.js

```diff
@@ -49,6 +63,49 @@ module.exports = function<T, P, I>(config : HostConfig<T, P, I>) : Reconciler {
 
   let nextUnitOfWork : ?Fiber = null;
 
+  let currentRootsWithPendingWork : ?Fiber = null;
+
+  function findNextUnitOfWork(priorityLevel : PriorityLevel) : ?Fiber {
+    let current = currentRootsWithPendingWork;
+    while (current) {
+      if (current.pendingWorkPriority !== 0 &&
+          current.pendingWorkPriority <= priorityLevel) {
+        // This node has work to do that fits our priority level criteria.
+        if (current.pendingProps !== null) {
+          // We found some work to do. We need to return the "work in progress"
+          // of this node which will be the alternate.
+          return current.alternate;
```

Comment:
I was going to ask how .pendingProps gets set on the alternate but it looks like maybe it's not and you fixed that in 7df1cb9b207af2ed1594c794b8e9cb5d62e8cd10? I'll continue reading with that assumption…